benchmarks/README.md

```diff
@@ -0,0 +1,88 @@
+
+These benchmarks depend upon you to create the simulation files to be benchmarked. For a certain simulation named ```example```, include a file in ```src/example/example.jl```. Always start the file with the following, as it provides you access to helper functions located in ```src/helpers```.
+```julia
+using DrWatson
+@quickactivate :benchmarks
+```
+
+Additionally, this file should contain the following functions.
+
+0. ```setup_config```, which will take in a ```Dict{String, Any}``` with the provided parameter names pointing to that task's provided configuration values. It is up to you to take these values, process them and then organization them to be passed along.
+1. ```setup_benchmark```, which will create the Decapode and run ```eval(gensim())``` on it. Return the evaluated function.
+2. ```create_mesh```, which will create the mesh upon which the simulation will run and also initialize the initial conditions and any constants/parameters. Return the mesh, initial conditions and constants/parameters in that order.
+3. ```create_simulate```, which will take the generated mesh and evaluated function and run the simulate function. Return the resulting function.
+4. ```run_simulation```, which will take the resulting simulation function, initial conditions and constants/parameters and run the solve. Return the result of the solve.
+
+## Running the Benchmarks
+
+Simply run ```scripts/main.sh``` and provide it the name of the desired simulation to benchmark. Note this must match the name provided in the user's config.
+
+**Warning**: The caller of this script should be in the `benchmarks` directory.
+
+Note that it may be the case that a benchmark run will hit an Out-of-Memory error. If that occurs, then you may update the reserved memory in ```scripts/array.sh```. Similarly, you may increase the time requested to run the benchmark.
+
+## Data Collection and Processing
+
+Once a simulation is run, their outputs will be saved in ```data/sims/"sim_name"```. The benchmark JSON files will contain the full result of the benchmarking run while the stat JLD2 files will contain the solve result statistics from DEStats.
+
+These files will then be processed and have aggregated data stored in ```data/exp_pro/"sim_name"/"slurm_job_id"/autogen```. Each result file in that directory will have the entire results from one single task. Because these files are expected to be collected using DrWatson's ```collect_results``` on the ```autogen``` directory, there is no intended correlation between a task and its result file. 
+
+**Warning**: Note that not all information from the benchmarking run is saved to the result files and any files in ```data/sims/"sim_name"``` will be deleted upon the next benchmark run. On the other hand, result files in the ```autogen``` directory mentioned before will never be deleted by the benchmarking.
+
+An example Markdown file is output in ```data/exp_pro/"sim_name"/"slurm_job_id"``` for user inspection, called ```final.md```.
```

Comment:
What does `exp_pro` stand for? A more informative name would be helpful here.

Reply:
`exp_pro` was a default folder provided by DrWatson, meant for "Data from processing experiments" per their docs. But it's fair enough to consider that we don't need to actually use these folders.